### ISSUE TITLE
Recursor 4.1.0rc2 Webserver doesn't do Webserver thing

### ISSUE THREAD

**Issue by jpmens** [PowerDNS/pdns#5920]
- Program:  Recursor
 - Issue type: Bug report
### Short description
Recursor 4.1.0rc2 doesn't properly serve Web page when `webserver` is configured. Bits and pieces show up and console logs API errors.

* Chrome directly reports `Unauthorized`
* IE and Edge as well as Firefox do as below

### Environment

 - Operating system: SLES12SP2
 - Software version: 4.1.0rc2
 - Software source: compiled locally

### Steps to reproduce

```
webserver=yes
webserver-address=10.1.1.1
webserver-allow-from=10.1.0.0/16
webserver-password=bla
webserver-port=8082
api-key=abcde6c7637d6393b19f3def180a8f15
```

as soon as I attempt to go to the URL `http://x:bla@10.1.1.1:8082/` Recursor shows a portion of the page (e.g. logo) and nothing much else. Console endlessly logs:

```
Nov 07 09:53:06 HTTP Request "/api/v1/servers/localhost/statistics": Authentication by API Key failed
Nov 07 09:53:06 HTTP Request "/api/v1/servers/localhost": Authentication by API Key failed
Nov 07 09:53:06 HTTP Request "/jsonstat": Authentication by API Key failed
Nov 07 09:53:06 HTTP Request "/jsonstat": Authentication by API Key failed
Nov 07 09:53:06 HTTP Request "/jsonstat": Authentication by API Key failed
Nov 07 09:53:06 HTTP Request "/jsonstat": Authentication by API Key failed
Nov 07 09:53:07 HTTP Request "/api/v1/servers/localhost/statistics": Authentication by API Key failed
Nov 07 09:53:07 HTTP Request "/api/v1/servers/localhost": Authentication by API Key failed
Nov 07 09:53:08 HTTP Request "/api/v1/servers/localhost/statistics": Authentication by API Key failed
Nov 07 09:53:08 HTTP Request "/api/v1/servers/localhost": Authentication by API Key failed
Nov 07 09:53:09 HTTP Request "/api/v1/servers/localhost/statistics": Authentication by API Key failed
Nov 07 09:53:09 HTTP Request "/api/v1/servers/localhost": Authentication by API Key failed
Nov 07 09:53:10 HTTP Request "/api/v1/servers/localhost/statistics": Authentication by API Key failed
Nov 07 09:53:10 HTTP Request "/api/v1/servers/localhost": Authentication by API Key failed
Nov 07 09:53:11 HTTP Request "/api/v1/servers/localhost/statistics": Authentication by API Key failed
Nov 07 09:53:11 HTTP Request "/api/v1/servers/localhost": Authentication by API Key failed
Nov 07 09:53:11 HTTP Request "/jsonstat": Authentication by API Key failed
Nov 07 09:53:11 HTTP Request "/jsonstat": Authentication by API Key failed
Nov 07 09:53:11 HTTP Request "/jsonstat": Authentication by API Key failed
Nov 07 09:53:11 HTTP Request "/jsonstat": Authentication by API Key failed
Nov 07 09:53:12 HTTP Request "/api/v1/servers/localhost/statistics": Authentication by API Key failed
Nov 07 09:53:12 HTTP Request "/api/v1/servers/localhost": Authentication by API Key failed
Nov 07 09:53:13 HTTP Request "/api/v1/servers/localhost/statistics": Authentication by API Key failed
Nov 07 09:53:13 HTTP Request "/api/v1/servers/localhost": Authentication by API Key failed
Nov 07 09:53:14 HTTP Request "/api/v1/servers/localhost/statistics": Authentication by API Key failed
Nov 07 09:53:14 HTTP Request "/api/v1/servers/localhost": Authentication by API Key failed
Nov 07 09:53:15 HTTP Request "/api/v1/servers/localhost/statistics": Authentication by API Key failed
Nov 07 09:53:15 HTTP Request "/api/v1/servers/localhost": Authentication by API Key failed
Nov 07 09:53:16 HTTP Request "/api/v1/servers/localhost/statistics": Authentication by API Key failed
Nov 07 09:53:16 HTTP Request "/api/v1/servers/localhost": Authentication by API Key failed
Nov 07 09:53:16 HTTP Request "/jsonstat": Authentication by API Key failed
Nov 07 09:53:16 HTTP Request "/jsonstat": Authentication by API Key failed
Nov 07 09:53:16 HTTP Request "/jsonstat": Authentication by API Key failed
Nov 07 09:53:16 HTTP Request "/jsonstat": Authentication by API Key failed
Nov 07 09:53:17 HTTP Request "/api/v1/servers/localhost/statistics": Authentication by API Key failed
Nov 07 09:53:17 HTTP Request "/api/v1/servers/localhost": Authentication by API Key failed
Nov 07 09:53:17 HTTP Request "/api/v1/servers/localhost/statistics": Authentication by API Key failed
Nov 07 09:53:17 HTTP Request "/api/v1/servers/localhost": Authentication by API Key failed
Nov 07 09:53:18 HTTP Request "/api/v1/servers/localhost/statistics": Authentication by API Key failed
Nov 07 09:53:18 HTTP Request "/api/v1/servers/localhost": Authentication by API Key failed
Nov 07 09:53:19 HTTP Request "/api/v1/servers/localhost/statistics": Authentication by API Key failed

```

API access with the API key work.

### Expected behaviour
I expect the Web server to show its statistics.

**Comment by jpmens** [comment]
![ffox-recursor](https://user-images.githubusercontent.com/60706/32488076-c169870e-c3ab-11e7-9484-fa132a702cec.png)

**Comment by paddg** [comment]
This two characters are "no". This password field accepts the api-key. The URL `http://ip:port` works here with Chrome. A popup appears where I can give the webserver-passord. Username doesn't matter.

**Comment by jpmens** [comment]
Chrome (Version 61.0.3163.100 (Official Build) (64-bit)) on OS/X and Safari (also OS/X) both prompt for authentication and then behave as in above screenshot. Only after I paste API key into `Password` field and hit enter, is the Web page shown. 

Also tested with `pdns-recursor-0.0.1843gf1f3f11-1pdns.el7.x86_64.rpm`

**Comment by aerique** [comment]
@jpmens If I understand you correctly your issue is that no statistics are shown until you paste the API key in the `API Key` form on the webpage served by the PDNS Recursor webserver?

**Comment by jpmens** [comment]
1. Chrome on Windows doesn't prompt for auth here (`61.0.3163.79 (Official Build) (64-bit)`)
2. Correct: no stats shown until API key is pasted into Password field.

**Comment by jpmens** [comment]
If entering API key where it says "Password" is intended behavior, I would suggest to label the field `API Key`.

**Comment by rgacogne** [comment]
> If entering API key where it says "Password" is intended behavior, I would suggest to label the field `API Key`.

@wojas  fixed it in #5925 :-)

**Comment by aerique** [comment]
After discussion on IRC we decided to close this issue since needing to enter the API key before the webpage shows statistics is intended behavior.

We need to improve the documentation and we'll make a separate ticket for the reported Chrome issues.
